src/Data/BTree/Primitives.hs

```diff
@@ -61,6 +89,16 @@ data Tree key val where
 deriving instance (Show key, Show val) => Show (Node height key val)
 deriving instance (Show key, Show val) => Show (Tree key val)
 
+instance (Binary key, Binary val) => Binary (Tree key val) where
+    put (Tree height rootId) = put height >> put rootId
+    get = Tree <$> get <*> get
+
+instance (Typeable key, Typeable val, Eq key, Eq val) => Eq (Tree key val) where
```

Comment:
since they are only used for testing, I've created `...Eq...` functions in the corresponding modules, to not clutter up the library. See fe542df